Core/include/Acts/Vertexing/TrackAtVertex.hpp

```diff
@@ -38,6 +38,8 @@ struct InputTrack {
     return m_ptr == other;
   }
 
+  bool operator<(const InputTrack& other) const { return m_ptr < other.m_ptr; }
```

Comment:
Would it make sense to make this spaceship as well?

Reply:
I thought about, but in the end, we are not using the other comparison modes. I think we only had it as a private friend to enable sorting or so.

I also thought about in some cases to replace everything (including the `operator==`) with
`auto operator<=>(const InputTrack& other) const = default;`
and deduce everything.
But this is not robust in sense of reordering and addition/removal of members, so maybe would require extra tests to guarantee a constant behaviour.